cfe_internal/update/simple_policy_update_v1.cf

```diff
@@ -0,0 +1,476 @@
+# are maintained by CFEngine and its original state is required for internal
+# operations of CFEngine. If the file has been modified CFEngine’s upgrades
+# may require manual intervention. Contact CFEngine support if additional
+# information and/or recommendation is needed.
+#
+################################################################################
+#########################################################
+#
+# update_policy
+# - Cfengine policy update (masterfiles -> inputs)
+#
+#########################################################
+
+bundle agent simple_policy_update_v1_cfe_internal_update_policy
+{
+  vars:
+    !cfengine_3_5::
+      "inputs_dir"         string => translatepath("$(sys.inputdir)"),
+      comment => "Directory containing CFEngine policies",
+      handle => "cfe_internal_update_policy_vars_inputs_dir";
+
+      "dir_bin"         string => translatepath("$(sys.bindir)"),
+      comment => "Directory containing CFEngine binaries",
+      handle => "cfe_internal_update_policy_vars_dir_bin";
+
+    cfengine_3_5::
+      "inputs_dir"         string => translatepath("$(sys.workdir)/inputs"),
+      comment => "Directory containing CFEngine policies",
+      handle => "cfe_internal_update_policy_vars_inputs_dir_backport";
+
+      "dir_bin"         string => translatepath("$(sys.workdir)/bin"),
+      comment => "Directory containing CFEngine binaries",
+      handle => "cfe_internal_update_policy_vars_dir_bin_backport";
+
+    windows::
+
+      "master_location"    string => "$(update_def.masterfiles_location)",  # NB! NOT $(sys.workdir) on Windows !
+      comment => "The master CFEngine policy directory on the policy host",
+      handle => "cfe_internal_update_policy_vars_master_location_windows";
+
+      "modules_dir"        string => "/var/cfengine/modules",      # NB! NOT $(sys.workdir) on Windows !
+      comment => "Directory containing CFEngine modules",
+      handle => "cfe_internal_update_policy_vars_modules_dir_windows";
+
+      "plugins_dir"        string => "/var/cfengine/plugins",      # NB! NOT $(sys.workdir) on Windows !
+      comment => "Directory containing CFEngine plugins",
+      handle => "cfe_internal_update_policy_vars_plugins_dir_windows";
+
+    !windows.!cfengine_3_5::
+      "master_location"
+         string => "$(cfe_internal_update_phase_and_channel_selection.masterfiles_location)",
+         ifvarclass => "cfe_internal_enable_phase_policy_channel",
+         comment => "The master CFEngine policy directory on the policy host",
+         handle => "cfe_internal_update_policy_vars_master_location_if_phase_based_policy_channel";
+
+      "master_location"
+         string =>  "$(update_def.masterfiles_location)",
+         ifvarclass => "!cfe_internal_enable_phase_policy_channel",
+         comment => "The master CFEngine policy directory on the policy host",
+         handle => "cfe_internal_update_policy_vars_master_location_if_not_phase_based_policy_channel";
+
```

Comment:
See that master_location is set based on cfe_internal_enable_phase_policy_channel. If the class is set, master_location gets sourced from the phase selection bundle, else it gets sourced from update_def. (This was a seperate improvement. I was positioning for being able to set it from def.json.)